tools/sigclient/pkg/query/functional.go

```diff
@@ -0,0 +1,567 @@
+		log.Fatalf("ReadAndValidateQueryFile: expectedResult not found")
+	}
+
+	expectedResult, isMap := response["expectedResult"].(map[string]interface{})
+	if !isMap {
+		log.Fatalf("ReadAndValidateQueryFile: expectedResult is not valid it's of type: %T", response["expectedResult"])
+	}
+
+	expRes := CreateResult(expectedResult)
+	if expRes.Qtype == "logs-query" {
+		populateTotalMatchedAndRecords(expectedResult, expRes)
+		if len(expRes.Records) == 0 || len(expRes.Records) > 10 {
+			log.Fatalf("ReadQueryFile: Number of records should be in range 1-10 for logs-query, got: %v", len(expRes.Records))
+		}
+	}
+
+	return query, expRes
+}
+
+// Main function that tests all the queries
+func FunctionalTest(dest string, dataPath string) {
+
+	queryFiles, err := os.ReadDir(dataPath)
+	if err != nil {
+		log.Fatalf("FunctionalTest: Error reading directory: %v, err: %v", dataPath, err)
+		return
+	}
+
+	// validate JSON files
+	for _, file := range queryFiles {
+		if !strings.HasSuffix(file.Name(), ".json") {
+			log.Fatalf("FunctionalTest: Invalid file format: %v. Expected .json", file.Name())
+			return
+		}
+	}
+
+	// Default values
+	startEpoch := "now-1h"
+	endEpoch := "now"
+	queryLanguage := "Splunk QL"
+
+	// run query
+	queryReq := map[string]interface{}{
+		"state":         "query",
+		"searchText":    "",
+		"startEpoch":    startEpoch,
+		"endEpoch":      endEpoch,
+		"indexName":     "*",
+		"queryLanguage": queryLanguage,
+	}
+
+	// run queries
+	for idx, file := range queryFiles {
+		filePath := filepath.Join(dataPath, file.Name())
+		query, expRes := ReadAndValidateQueryFile(filePath)
+		log.Infof("FunctionalTest: qid=%v, Running query=%v", idx, query)
+		queryReq["searchText"] = query
+
+		EvaluateQueryForWebSocket(dest, queryReq, idx, expRes)
+		EvaluateQueryForAPI(dest, queryReq, idx, expRes)
+
+		log.Infoln()
+	}
+
+	log.Infof("FunctionalTest: All queries passed successfully")
+}
+
+func EvaluateQueryForAPI(dest string, queryReq map[string]interface{}, qid int, expRes *Result) {
+	query := queryReq["searchText"].(string)
+
+	reqBody, err := json.Marshal(queryReq)
+	if err != nil {
+		log.Fatalf("EvaluateQueryForAPI: Error marshaling request: %v", err)
+	}
+
+	url := fmt.Sprintf("http://%s/api/search", dest)
+	req, err := http.NewRequestWithContext(context.TODO(), http.MethodPost, url, bytes.NewReader(reqBody))
+	if err != nil {
+		log.Fatalf("EvaluateQueryForAPI: Error creating request: %v", err)
+	}
+	req.Header.Set("content-type", "application/json")
+
+	sTime := time.Now()
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		log.Fatalf("EvaluateQueryForAPI: Error sending request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	responseData := make(map[string]interface{})
+	if bodyBytes, err := io.ReadAll(resp.Body); err != nil {
+		log.Fatalf("EvaluateQueryForAPI: Error reading response: %v", err)
+	} else {
+		if err := json.Unmarshal(bodyBytes, &responseData); err != nil {
+			if len(bodyBytes) > 20 {
```

Comment:
what is this magical number 20 ?

Reply:
It was just to reduce the amount of data to be logged. Removed this and now logging the whole body. Addressed in the latest commit.